### PR TITLE
feat: transfer process fragment migrated to mvvm

### PR DESCRIPTION
**Pull request by PratyushSingh07**

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
hey can you review this @gururani-abhishek

**Comment by PratyushSingh07** [comment]
hey @jawidMuhammadi can you please review this ?

**Comment by PratyushSingh07** [comment]
The merge conflicts have been solved @jawidMuhammadi

File: app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt [hunk 1]
```diff
@@ -49,42 +47,128 @@ class TransferProcessFragment : BaseFragment(), TransferProcessView {
...
+        val transferAmount = binding.tvAmount.text.toString().toDouble()
+        val transferDescription = binding.tvRemark.text.toString()
+        var dateFormat: String = "dd MMMM yyyy"
+        var locale: String = "en"
+        val fromAccountNumber = fromAccountOption?.accountNo
+        val toAccountNumber = toAccountOption?.accountNo
         if (transferType == TransferType.SELF) {
-            presenter?.makeSavingsTransfer(payload)
+            viewModel.makeSavingsTransfer(
+                fromOfficeId,
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt]: Do we need to create a variable, what about if we pass the value directly like this:

```
viewModel.makeSavingsTransfer(
     fromAccountOption?.officeId,
    ...
}
```

File: app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt [hunk 2]
```diff
@@ -49,42 +47,128 @@ class TransferProcessFragment : BaseFragment(), TransferProcessView {
...
+                toAccountId,
+                transferDate,
+                transferAmount,
+                transferDescription,
+                dateFormat,
+                locale,
+                fromAccountNumber,
+                toAccountNumber
+            )
         } else if (transferType == TransferType.TPT) {
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt]: Why do we need to check for the `transferType` here? However, in the repository I see we are calling the same method for both of them. It is better to do this kind of checking in the data layer.

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt]: done

Comment (jawidMuhammadi) [reply app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt]: Are you sure? :)

Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt]: my bad , forgot to push the code :(

**Comment by PratyushSingh07** [comment]
solved the conflicts and pushed the updated code @jawidMuhammadi 
PS : I didnt forget to push it this time

**Pull request closed without merging**